docs/whatsnew/6.1.rst

```diff
@@ -43,6 +43,41 @@ columns with any long integers which overflowed the 32-bit integer would be retu
 as string columns. The new default behavior is consistent with ``numpy`` v2 and ``pandas``.
 
 
+Updates to `~astropy.cosmology`
+===============================
+
+:class:`~astropy.cosmology.Cosmology` as a :func:`~dataclasses.dataclass`
+-------------------------------------------------------------------------
+
+The :class:`~astropy.cosmology.Cosmology` class is
+now a :func:`~dataclasses.dataclass`. This means that the :mod:`dataclasses` machinery
+can be used to work with :class:`~astropy.cosmology.Cosmology` objects. For example::
+
+.. doctest-skip::
+
+    >>> from dataclasses import asdict, fields, replace
+    >>> from astropy.cosmology import Planck18
+    >>> replace(Planck18, name="modified", Ob0=0.05)
+    FlatLambdaCDM(name='modified', ..., Ob0=0.05)
+    >>> asdict(Planck18)
+    {'name': 'Planck18', 'meta': ..., 'H0': <Quantity 67.66 km / (Mpc s)>, ...
+    >>> [f.name for f in fields(Planck18)]
+    ['name', 'meta', 'H0', 'Om0', 'Ode0', 'Tcmb0', 'Neff', 'm_nu', 'Ob0']
+
+
+Also, it is now possible to create new :class:`~astropy.cosmology.Cosmology` subclasses
+using :func:`~dataclasses.make_dataclass`::
+
+.. doctest-skip::
+
+    >>> from dataclasses import make_dataclass, field, fields
+    >>> from astropy.cosmology import Cosmology
+    >>> NewC = make_dataclass("NewC", [("newfield", float, field(default=None))],
+    ...                       bases=(Cosmology,), frozen=True, eq=False)
+    >>> [f.name for f in fields(NewC)]
+    [..., 'newfield']
```

Comment:
Are there too many fields to write out all of them? If some fields should be skipped then I'd rather keep the fields in the beginning and end and only skip the ones in the middle.

Reply:
It was just to emphasize `'newfield'`.

Reply:
I added the other fields.